core_types/src/crud_status.rs

```diff
@@ -24,52 +24,38 @@ pub fn create_crud_link_eav(from: &Address, to: &Address) -> EntityAttributeValu
     EntityAttributeValue::new(from, &LINK_NAME.to_string(), to)
 }
 
-bitflags! {
-    #[derive(Default, Serialize, Deserialize, DefaultJson)]
-    /// the CRUD status of a Pair is stored as EntryMeta in the hash table, NOT in the entry itself
-    /// statuses are represented as bitflags so we can easily build masks for filtering lookups
-    pub struct CrudStatus: u8 {
-        const LIVE = 0x01;
-        const REJECTED = 0x02;
-        const DELETED = 0x04;
-        const MODIFIED = 0x08;
-        /// CRDT resolution in progress
-        const LOCKED = 0x10;
-    }
+/// the CRUD status of a Pair is stored as EntryMeta in the hash table, NOT in the entry itself
+/// statuses are represented as bitflags so we can easily build masks for filtering lookups
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize, DefaultJson)]
+#[serde(rename_all = "lowercase")]
+pub enum CrudStatus {
+    Live,
+    Rejected,
+    Deleted,
+    Modified,
+    /// CRDT resolution in progress
+    Locked,
 }
 
-impl From<CrudStatus> for String {
-    fn from(crud_status: CrudStatus) -> String {
-        // don't do self.bits().to_string() because that spits out values for default() and all()
-        // only explicit statuses are safe as strings
-        // the expectation is that strings will be stored, referenced and parsed
-        String::from(match crud_status {
-            CrudStatus::LIVE => "1",
-            CrudStatus::REJECTED => "2",
-            CrudStatus::DELETED => "4",
-            CrudStatus::MODIFIED => "8",
-            CrudStatus::LOCKED => "16",
-            _ => unreachable!(),
-        })
+impl Default for CrudStatus {
+    fn default() -> CrudStatus {
+        CrudStatus::Live
     }
 }
 
-impl From<&'static str> for CrudStatus {
-    fn from(s: &str) -> CrudStatus {
-        CrudStatus::from(String::from(s))
+impl From<CrudStatus> for String {
+    fn from(status: CrudStatus) -> String {
+        let res_str = serde_json::to_string(&status).expect("failed to serialize CrudStatus enum");
```

Comment:
if it fails should we be utilzing try_from instead?

Reply:
No, usually I would implement the designated `FromStr` trait which was specifically made for this.